frontend/controller/app/group.js

```diff
@@ -28,9 +28,11 @@ sbp('okTurtles.events/on', LEFT_GROUP, ({ identityContractID, groupContractID })
   // grab the groupID of any group that we're a part of
   const currentGroupId = rootState.currentGroupId
   if (!currentGroupId || currentGroupId === groupContractID) {
-    const groupIdToSwitch = Object.keys(state.groups)
+    const groupIdToSwitch = Object.entries(state.groups)
+      // $FlowFixMe[incompatible-use]
+      .map(([cID, { hasLeft }]) => !hasLeft && cID)
       .filter(cID =>
-        cID !== groupContractID
+        cID && cID !== groupContractID
```

Comment:
Is this `cID &&` business necessary? You already do `&& cID` in the `.map`...

Reply:
It is, because it could be `null` / `undefined` / `false`, &c.

Reply:
Because of the `cID !== groupContactID` part it could be fine without (if we're use `groupContractID` is never falsy), but I think explicit is good because it lowers the cognitive load to understand exactly what's going on.